homeworkWeekend10/src/ba/bitcamp/SQL/task02andtask03/SQLConnection.java

```diff
@@ -0,0 +1,185 @@
+			JOptionPane.showMessageDialog(null, "Inputed code does not have 10 digits.");
+			isInputed = false;
+			return;
+		}
+		// Checking if the all characters of the code are digits.
+		for (int i = 0; i < code.length(); i++) {
+			if (!Character.isDigit(code.charAt(i))) {
+				JOptionPane.showMessageDialog(null, "Please for code input only digits.");
+				isInputed = false;
+				return;
+			}
+		}
+		// Calling method checkProduct and checking if there is the product in
+		// the database with the same name.
+		if (!checkProduct(name).equals("")) {
+			int option = JOptionPane.showConfirmDialog(null, "The product already exists in database. Do you want to continue?", "", JOptionPane.YES_NO_OPTION);
+			if (option == JOptionPane.NO_OPTION) {
+				isInputed = false;
+				return;
+			}
+		}
+
+		try {
+			// Declaring and executing query.
+			Statement stmt = conn.createStatement();
+			stmt.executeUpdate("INSERT INTO product VALUES ('" + code + "', '" + name + "', " + price + ")");
+			JOptionPane.showMessageDialog(null, "You have successfully inputed the product.");
+			System.out.println("Query executed.");
+			isInputed = true;
+		} catch (SQLException e) {
+			// If in the database already exists product with inputed code
+			// opening WARNING message.
+			JOptionPane.showMessageDialog(null, "The product with inputed code " + code + " already exist in database.");
+			return;
+		}
+	}
+
+	/**
+	 * Checks if there is the product with the same name in the database.
+	 * 
+	 * @param product
+	 *            - Name of the product.
+	 * @return - The empty string if there is no products with same name.
+	 */
+	public static String checkProduct(String product) {
+		// Declaring string variable.
+		StringBuilder sb = new StringBuilder();
+		try {
+			// Declaring and executing query.
+			Statement stmt = conn.createStatement();
+			ResultSet result = stmt.executeQuery("SELECT name FROM product WHERE name = '" + product + "'");
+
+			// Reading from database and saving into string variable.
+			while (result.next()) {
+				sb.append(result.getString(1) + "\n");
+			}
+		} catch (SQLException e) {
+			System.err.println("Not valid SQL command.");
+			System.err.println("Message: " + e.getMessage());
+			System.exit(1);
+		}
+		// Returning string variable.
+		return sb.toString();
+	}
+
+	/**
+	 * Reads the product from the database and saves them into the list of
+	 * products.
+	 * 
+	 * @return - The list of the products from database.
+	 */
+	public static ArrayList<Product> selectProducts() {
+		// Declaring the list of the products.
+		ArrayList<Product> array = new ArrayList<>();
+		try {
+			// Declaring and executing query.
+			Statement stmt = conn.createStatement();
+			ResultSet result = stmt.executeQuery("SELECT * FROM product");
+
+			// Reading from the database and inputing products into the list.
+			while (result.next()) {
+				array.add(new Product(result.getInt(1), result.getString(2), result.getDouble(3)));
```

Comment:
Instead of `hard-coding` column indexes, better practice is to use column names to retrieve values from result-set.